### PR TITLE
Added source ru.readmanga

### PR DESCRIPTION
**Pull request by leugenea**
Checklist:
- [x] Updated source's version for individual source changes
- [x] Updated all sources' versions for template changes
- [x] Set appropriate `nsfw` value
- [x] Did not change `id` even if a source's name or language were changed
- [x] Tested the modifications by running it on the simulator or a test device 
- [x] Run `cargo +nightly clippy`

### CODE REVIEW

**Comment by Skittyblock** [comment]
make sure to run `cargo +nightly clippy` and correct the warnings that it gives you.

**Comment by leugenea** [comment]
> make sure to run `cargo +nightly clippy` and correct the warnings that it gives you. 

Ran clippy, fixed warnings

File: src/rust/ru.readmanga/.gitignore
Comment (Skittyblock) [inline src/rust/ru.readmanga/.gitignore:1]: We shouldn't have a custom gitignore here, just don't stage your `.vscode` folder before committing

File: src/rust/ru.readmanga/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	// 	Node::new_fragment(b"<a href=\"foo\"><div>foo</div></a>")
+	// 		.unwrap()
+	// 		.outer_html()
+	// 		.read()
+	// );
+	// debug!(
+	// 	"{:?}",
+	// 	WNode::new("<a href=\"foo\"><div>foo</div></a>".to_string()).attr("href")
+	// );
+}
```
Comment (Skittyblock) [inline src/rust/ru.readmanga/src/lib.rs:36]: Remove this if it's not necessary

File: src/rust/ru.readmanga/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+#[get_manga_list]
+pub fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let search_url = parser::get_filter_url(&filters, &Sorting::default(), page)?;
+	let html = helpers::get_html(&search_url)?;
+	let mangas = parser::parse_search_results(&html)?;
+	let result = helpers::create_manga_page_result(mangas);
+	debug!("get_manga_list: {result:?}");
```
Comment (Skittyblock) [inline src/rust/ru.readmanga/src/lib.rs:44]: We shouldn't leave in any debug logging

File: src/rust/ru.readmanga/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	let url = helpers::get_chapter_url(&manga_id, &chapter_id);
+	let html = helpers::get_html(&url)?;
+	let result = parser::get_page_list(&html);
+	debug!("get_page_list: {result:?}");
+
+	result
+}
+
+#[modify_image_request]
+pub fn modify_image_request(_: Request) {}
```
Comment (Skittyblock) [inline src/rust/ru.readmanga/src/lib.rs:92]: Remove this if it's not necessary

File: src/rust/ru.readmanga/src/parser.rs [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+use crate::{
+	constants::{BASE_SEARCH_URL, BASE_URL, SEARCH_OFFSET_STEP},
+	get_manga_details, helpers,
+	sorting::Sorting,
+	wrappers::{debug, WNode},
+};
+
+pub fn parse_search_results(html: &WNode) -> Result<Vec<Manga>> {
+	let nodes = html.select("div.tile");
+	// debug!("{:?}", nodes);
```
Comment (Skittyblock) [inline src/rust/ru.readmanga/src/parser.rs:26]: Probably remove all these debug log comments

**Review by Skittyblock — CHANGES_REQUESTED** [review]

File: src/rust/ru.readmanga/.gitignore
Comment (leugenea) [reply src/rust/ru.readmanga/.gitignore:1]: Why? This gitignore is only for this folder

File: src/rust/ru.readmanga/src/lib.rs (hunk 2 above)
Comment (leugenea) [reply src/rust/ru.readmanga/src/lib.rs:44]: Why? Does it slow code much?

File: src/rust/ru.readmanga/.gitignore
Comment (leugenea) [reply src/rust/ru.readmanga/.gitignore:1]: Removed

File: src/rust/ru.readmanga/src/lib.rs (hunk 1 above)
Comment (leugenea) [reply src/rust/ru.readmanga/src/lib.rs]: Removed

File: src/rust/ru.readmanga/src/lib.rs (hunk 2 above)
Comment (leugenea) [reply src/rust/ru.readmanga/src/lib.rs]: Removed

File: src/rust/ru.readmanga/src/lib.rs (hunk 3 above)
Comment (leugenea) [reply src/rust/ru.readmanga/src/lib.rs]: Removed

File: src/rust/ru.readmanga/src/parser.rs (hunk 4 above)
Comment (leugenea) [reply src/rust/ru.readmanga/src/parser.rs]: Removed

File: src/rust/ru.readmanga/src/lib.rs (hunk 2 above)
Comment (Skittyblock) [reply src/rust/ru.readmanga/src/lib.rs]: Yeah, each call to an imported function is a little expensive, especially when it has to write and read strings in memory from the WASM into a Swift String instance. Also, it just pollutes the log.

**Comment by leugenea** [comment]
@Skittyblock , do I have to do any changes or everything is ok?

File: src/rust/ru.readmanga/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+[package]
+name = "readmanga"
+version = "1.0.3"
```
Comment (Skittyblock) [inline src/rust/ru.readmanga/Cargo.toml:3]: This is the initial version you're adding to the repo, so it should be `1.0.0` (that being said, the cargo package name/version aren't used and don't matter atm).

File: src/rust/ru.readmanga/res/source.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{
+	"info": {
+		"id": "ru.readmanga",
+		"lang": "ru",
+		"name": "ReadManga",
+		"version": 1,
+		"urls": [
+			"https://readmanga.live/"
```
Comment (Skittyblock) [inline src/rust/ru.readmanga/res/source.json:8]:
```suggestion
			"https://readmanga.live"
```

File: src/rust/ru.readmanga/res/source.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+		],
```
Comment (Skittyblock) [inline src/rust/ru.readmanga/res/source.json:9]:
```suggestion
		"url": "https://readmanga.live",
```

**Comment by Skittyblock** [comment]
https://readmanga.live gives me a 404. Is that because of my location?

File: src/rust/ru.readmanga/Cargo.toml (hunk 5 above)
Comment (leugenea) [reply src/rust/ru.readmanga/Cargo.toml]: Changed back to 1.0.0

**Comment by leugenea** [comment]
> https://readmanga.live gives me a 404. Is that because of my location?

I guess it is. But it opens from my internet provider and with VPN.
image

**Review by Skittyblock — APPROVED** [review]

**Pull request merged**